test/tools/jsontests/TransactionTests.cpp

```diff
@@ -38,134 +39,174 @@ namespace fs = boost::filesystem;
+		mObject obj;
+		obj = value.get_obj();
+		BOOST_REQUIRE_MESSAGE(obj.count("network"), "network section not set in expect section!");
+		vector<string> networks;
+		ImportTest::parseJsonStrValueIntoVector(obj["network"], networks);
+		BOOST_CHECK_MESSAGE(networks.size() > 0, TestOutputHelper::get().testName() + " Network array not set!");
+		ImportTest::checkAllowedNetwork(networks);
+
+		if (test::inArray(networks, test::netIdToString(_network)) || test::inArray(networks, string("ALL")))
+			objVector.push_back(obj);
+	}
+	BOOST_REQUIRE_MESSAGE(objVector.size() == 1, "Expect network should occur once in expect section of transaction test filler! (" + test::netIdToString(_network) + ")");
+	return objVector.at(0);
+}
+
+json_spirit::mObject FillTransactionTest(json_spirit::mObject const& _o)
+{
+	mObject out;
+	BOOST_REQUIRE_MESSAGE(_o.count("transaction") > 0, "transaction section not found! " + TestOutputHelper::get().testFileName());
+	BOOST_REQUIRE_MESSAGE(_o.count("expect") > 0, "expect section not found! " + TestOutputHelper::get().testFileName());
+
+	mObject tObj = _o.at("transaction").get_obj();
+	RLPStream rlpStream = createRLPStreamFromTransactionFields(tObj);
+	out["rlp"] = toHexPrefixed(rlpStream.out());
+
+	// Theoretical block for transaction check
+	BlockHeader bh;
+	bh.setNumber(1);	//Seal engine below enables network rules from block 0
+	bh.setGasLimit(u256("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
+
+	mValue expectObj = _o.at("expect");
+	for (auto const network: test::getNetworks())
+	{
+		ChainParams params(genesisInfo(network));
+		unique_ptr<SealEngineFace> se(params.createSealEngine());
+
+		// Test networks has forkblocks set to 0 if rules are enabled
+		bool onConstantinople = (bh.number() >= params.constantinopleForkBlock);
+		out[test::netIdToString(network)] = mObject();
+		mObject expectSection = getExpectSection(expectObj, network);
+		try
+		{
+			Transaction txFromFields(rlpStream.out(), CheckTransaction::Everything);
+			bool onConstantinopleAndZeroSig = onConstantinople && txFromFields.hasZeroSignature();
+			if (!(txFromFields.signature().isValid() || onConstantinopleAndZeroSig))
+				BOOST_THROW_EXCEPTION(Exception() << errinfo_comment(TestOutputHelper::get().testName() + "transaction from RLP signature is invalid") );
+
+			se->verifyTransaction(ImportRequirements::Everything, txFromFields, bh, 0);
+			if (expectSection.count("sender") > 0)
+			{
+				string expectSender = toString(expectSection["sender"].get_str());
+				BOOST_CHECK_MESSAGE(toString(txFromFields.sender()) == expectSender, "Error filling transaction test " + TestOutputHelper::get().testName() + ": expected another sender address! (got: " + toString(txFromFields.sender()) + "), expected: (" + expectSender + ")");
+			}
+
+			mObject resultObject;
+			resultObject["sender"] = toString(txFromFields.sender());
+			//resultObject["transaction"] = ImportTest::makeAllFieldsHex(tObj);
```

Comment:
Why is this commented out?

Reply:
I decided that it is not neccessary to print and check rlp decoded from string for each fork. 
this check should be done by hash.

Reply:
OK.

Reply:
should I remove the comment for good?

Reply:
Yes, I think so.